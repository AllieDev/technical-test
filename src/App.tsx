import { useState } from "react";
import reactLogo from "./assets/react.svg";
import Header from "./components/layouts/Header";
import Footer from "./components/layouts/Footer";
import Map from "./components/maps/Map";
import "./App.css";

function App() {
  const [count, setCount] = useState(0);

  return (
    <div className="App">
      <Header />
      <Map />
      <Footer />
    </div>
  );
}

export default App;
