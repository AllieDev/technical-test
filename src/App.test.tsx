import App from "./App";
import { describe, expect, it, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { store } from "./store";
import { Provider } from "react-redux";

vi.mock("./components/maps/Map", () => {
  return {
    default: () => <div>Mock Map</div>,
  };
});

describe("#App", () => {
  it("serial test", () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container.textContent).toMatchSnapshot();
  });
});
