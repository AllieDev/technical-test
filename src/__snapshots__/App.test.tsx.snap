// Vitest Snapshot v1

exports[`#App > serial test 1`] = `"Arcgis MapHomeAboutContactLoading..Mock MapÂ© COPYRIGHT ALI ASGHAR SULTANI - 2023"`;
